JitsiConferenceEventManager.js

```diff
@@ -0,0 +1,495 @@
+                conference.statistics.sendSetRemoteDescFailed(e, pc);
+            });
+
+        conference.room.addListener(XMPPEvents.ADD_ICE_CANDIDATE_FAILED,
+            function (e, pc) {
+                conference.statistics.sendAddIceCandidateFailed(e, pc);
+            });
+    }
+};
+
+/**
+ * Setups event listeners related to conference.rtc
+ */
+JitsiConferenceEventManager.prototype.setupRTCListeners = function () {
+    var conference = this.conference;
+    conference.rtc.addListener(RTCEvents.DOMINANTSPEAKER_CHANGED,
+        function (id) {
+            if(conference.lastDominantSpeaker !== id && conference.room) {
+                conference.lastDominantSpeaker = id;
+                conference.eventEmitter.emit(
+                    JitsiConferenceEvents.DOMINANT_SPEAKER_CHANGED, id);
+            }
+            if (conference.statistics && conference.myUserId() === id) {
+                // We are the new dominant speaker.
+                conference.statistics.sendDominantSpeakerEvent();
+            }
+        });
+
+    conference.rtc.addListener(RTCEvents.DATA_CHANNEL_OPEN, function () {
+        var now = window.performance.now();
+        logger.log("(TIME) data channel opened ", now);
+        conference.room.connectionTimes["data.channel.opened"] = now;
+    });
+
+    conference.rtc.addListener(RTCEvents.LASTN_CHANGED,
+        function (oldValue, newValue) {
+            conference.eventEmitter.emit(
+                JitsiConferenceEvents.IN_LAST_N_CHANGED, oldValue, newValue);
+        });
+
+    conference.rtc.addListener(RTCEvents.LASTN_ENDPOINT_CHANGED,
+        function (lastNEndpoints, endpointsEnteringLastN) {
+            conference.eventEmitter.emit(
+                JitsiConferenceEvents.LAST_N_ENDPOINTS_CHANGED,
+                lastNEndpoints, endpointsEnteringLastN);
+        });
+
+    conference.rtc.addListener(RTCEvents.AVAILABLE_DEVICES_CHANGED,
+        function (devices) {
+            conference.room.updateDeviceAvailability(devices);
+        });
+};
+
+/**
+ * Setups event listeners related to conference.xmpp
+ */
+JitsiConferenceEventManager.prototype.setupXMPPListeners = function () {
+    var conference = this.conference;
+    conference.xmpp.addListener(
+        XMPPEvents.CALL_INCOMING, conference.onIncomingCall.bind(conference));
+
+    conference.xmpp.addListener(XMPPEvents.START_MUTED_FROM_FOCUS,
+        function (audioMuted, videoMuted) {
+            conference.startAudioMuted = audioMuted;
+            conference.startVideoMuted = videoMuted;
+
+            // mute existing local tracks because this is initial mute from
+            // Jicofo
+            conference.getLocalTracks().forEach(function (track) {
+                if (conference.startAudioMuted && track.isAudioTrack()) {
+                    track.mute();
+                }
+                if (conference.startVideoMuted && track.isVideoTrack()) {
+                    track.mute();
+                }
```

Comment:
The above conditions seem susceptible to improvement to me because isAudioTrack precludes the possibility of satisfying isVideoTrack as well. I guess 1 could reason 2 ways:

1. Put the isAudioTrack/isVideoTrack first:
````javascript
if (track.isAudioTrack()) {
    conference.startAudioMuted && track.mute();
} else if (track.isVideoTrack()) {
    conference.startVideoMuted && track.mute();
}
````

Of course, one could reason that isAudioTrack/isVideoTrack is slower (because it's more complex a function) than the booleans such as startAudioMuted & startVideoMuted. Since we're in lib-jitsi-meet, we could improve that drawback with a switch on track.getType() because there'll be 1 getType() call:
````javascript
switch (track.getType()) {
case MediaType.AUDIO:
    conference.startAudioMuted && track.mute();
    break;
case MediaType.VIDEO:
    conference.startVideoMuted && track.mute();
    break;
}
````

2. As an alternative to 1. & if we don't want to sacrifice too much, we could at least improve by noticing that if the first if is satisfied, then there's no point in checking the second if. That could be solved with a return in the first if or converting the 2 if to if-else-if.